azurerm/internal/services/containers/container_registry_scope_map_data_source.go

```diff
@@ -0,0 +1,74 @@
+package containers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceContainerRegistryScopeMap() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceContainerRegistryScopeMapRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
since we have a validation function for the name can we use it here to?

Reply:
There are system scope maps that uses a different naming standard so we cannot use the same validation for the data source.